winmake.ps1

```diff
@@ -31,44 +31,15 @@ function Local-Machine {
 # Expect starting directory to be /podman
 function Win-SSHProxy {
     param (
-        [string]$Ref
+        [string]$Version
     );
 
-    # git is not installed by default on windows
-    # fail if git doesn't exist
-    Get-Command git -ErrorAction SilentlyContinue  | out-null
-    if(!$?){
-        Write-Host "Git not installed, cannot build Win-SSHProxy"
-        Exit 1
-    }
-
-    if (Test-Path ./tmp-gv) {
-        Remove-Item ./tmp-gv -Recurse -Force -Confirm:$false
-    }
-
-    $GV_GITURL = "https://github.com/containers/gvisor-tap-vsock.git"
-
-    New-Item  ./tmp-gv -ItemType Directory -ea 0
-    Push-Location ./tmp-gv
-    Run-Command "git init"
-    Run-Command "git remote add origin $GV_GITURL"
-    Run-Command "git fetch --depth 1 origin main"
-    Run-Command "git fetch --depth 1 --tags"
-    Run-Command "git checkout main"
-    if (-Not $Ref) {
-        Write-Host "empty"
-        $Ref = git describe --abbrev=0
-    }
-    Run-Command "git checkout $Ref"
-    Run-Command "go build -ldflags -H=windowsgui -o bin/win-sshproxy.exe ./cmd/win-sshproxy"
-    Run-Command "go build -ldflags -H=windowsgui -o bin/gvproxy.exe ./cmd/gvproxy"
-    Pop-Location
-
-    # Move location to ./bin/windows for packaging script and for Windows binary testing
     New-Item -ItemType Directory -Force -Path "./bin/windows"
-    Copy-Item -Path "tmp-gv/bin/win-sshproxy.exe" -Destination "./bin/windows/"
-    Copy-Item -Path "tmp-gv/bin/gvproxy.exe" -Destination "./bin/windows/"
-    Remove-Item ./tmp-gv -Recurse -Force -Confirm:$false
+    if (-Not $Version) {
+        $Version = "v0.7.1"
+    }
+    Invoke-WebRequest -Uri "https://github.com/containers/gvisor-tap-vsock/releases/download/$Version/gvproxy-windowsgui.exe" -OutFile "./bin/windows/gvproxy.exe"
```

Comment:
Invoke-WebRequest is very slow unless you disable progress preference (adding $ProgressPreference = “SilentlyContinue” just before). Another issue is you often need to set UseBasicParsing, as otherwise it tries to use components of internet explorer which may not be active. Alternatively all  modern versions of  Windows have curl installed but you have to address it as “curl.exe” as “curl” becomes an alias of Invoke-WebRequest

Reply:
Thanks for the hint, I've switched to using curl.exe, this makes the Makefile and winmake changes almost the same!